### ISSUE TITLE
TS0601 TRV exposes a lot of switches

### ISSUE THREAD

**Issue by itavero** [Koenkk/zigbee-herdsman-converters#2302]
Someone filed a bug report for my Homebridge plugin ( itavero/homebridge-z2m#79 ) and I was very surprised to see that the `switch` type was used in the [Exposes](https://www.zigbee2mqtt.io/information/exposes) information here.

It was especially strange to see a lot of switches all on the same endpoint and after looking at the Exposes docs on the Zigbee2MQTT website, I wonder of this information is according to "spec".

The things that are being controlled (`window_detection`, `valve_detection`, `away_mode`, `auto_lock`) using these "switches" all seem to be related to the `climate` device it exposes and I would have probably expected these to be part of `features` of the `climate` entry with type `binary`.

On the other hand, my interpretation of what a "switch" is might be wrong, as I've always considered it to be something that turns on/off a device (like a light switch).

I think this is also the first device that I've come across that has the same type multiple times on a single endpoint. I wasn't aware that this is expected. From a "programming" point of view it's also not very convenient, perhaps even problematic. How is the software supposed to know how to distinguish these?

**Comment by itavero** [comment]
Copied the exposes information of the device from the bug report here for reference:

```json
{
   "date_code":"",
   "definition":{
      "description":"Radiator valve with thermostat",
      "exposes":[
         {
            "features":[
               {
                  "access":3,
                  "description":"Enables/disables physical input on the device",
                  "name":"state",
                  "property":"child_lock",
                  "type":"binary",
                  "value_off":"UNLOCK",
                  "value_on":"LOCK"
               }
            ],
            "type":"lock"
         },
         {
            "features":[
               {
                  "access":3,
                  "description":"Enables/disables window detection on the device",
                  "name":"state",
                  "property":"window_detection",
                  "type":"binary",
                  "value_off":"OFF",
                  "value_on":"ON",
                  "value_toggle":"TOGGLE"
               }
            ],
            "type":"switch"
         },
         {
            "access":1,
            "description":"Indicates if the battery of this device is almost empty",
            "name":"battery_low",
            "property":"battery_low",
            "type":"binary",
            "value_off":false,
            "value_on":true
         },
         {
            "features":[
               {
                  "access":3,
                  "name":"state",
                  "property":"valve_detection",
                  "type":"binary",
                  "value_off":"OFF",
                  "value_on":"ON",
                  "value_toggle":"TOGGLE"
               }
            ],
            "type":"switch"
         },
         {
            "access":1,
            "description":"Position",
            "name":"position",
            "property":"position",
            "type":"numeric",
            "unit":"%"
         },
         {
            "features":[
               {
                  "access":3,
                  "description":"Temperature setpoint",
                  "name":"current_heating_setpoint",
                  "property":"current_heating_setpoint",
                  "type":"numeric",
                  "unit":"°C",
                  "value_max":35,
                  "value_min":5,
                  "value_step":0.5
               },
               {
                  "access":1,
                  "description":"Current temperature measured on the device",
                  "name":"local_temperature",
                  "property":"local_temperature",
                  "type":"numeric",
                  "unit":"°C"
               },
               {
                  "access":3,
                  "description":"Mode of this device",
                  "name":"system_mode",
                  "property":"system_mode",
                  "type":"enum",
                  "values":[
                     "heat",
                     "auto",
                     "off"
                  ]
               },
               {
                  "access":1,
                  "description":"The current running state",
                  "name":"running_state",
                  "property":"running_state",
                  "type":"enum",
                  "values":[
                     "idle",
                     "heat"
                  ]
               },
               {
                  "access":3,
                  "description":"Offset to be used in the local_temperature",
                  "name":"local_temperature_calibration",
                  "property":"local_temperature_calibration",
                  "type":"numeric",
                  "unit":"°C"
               },
               {
                  "access":3,
                  "description":"Away mode",
                  "name":"away_mode",
                  "property":"away_mode",
                  "type":"binary",
                  "value_off":"OFF",
                  "value_on":"ON"
               },
               {
                  "access":3,
                  "description":"Mode of this device (similar to system_mode)",
                  "name":"preset",
                  "property":"preset",
                  "type":"enum",
                  "values":[
                     "schedule",
                     "manual",
                     "boost",
                     "complex",
                     "comfort",
                     "eco"
                  ]
               }
            ],
            "type":"climate"
         },
         {
            "features":[
               {
                  "access":3,
                  "description":"Enable/disable auto lock",
                  "name":"state",
                  "property":"auto_lock",
                  "type":"binary",
                  "value_off":"MANUAL",
                  "value_on":"AUTO"
               }
            ],
            "type":"switch"
         },
         {
            "features":[
               {
                  "access":3,
                  "description":"Enable/disable away mode",
                  "name":"state",
                  "property":"away_mode",
                  "type":"binary",
                  "value_off":"OFF",
                  "value_on":"ON"
               }
            ],
            "type":"switch"
         },
         {
            "access":3,
            "description":"Away preset days",
            "name":"away_preset_days",
            "property":"away_preset_days",
            "type":"numeric"
         },
         {
            "access":3,
            "description":"Boost time",
            "name":"boost_time",
            "property":"boost_time",
            "type":"numeric",
            "unit":"s"
         },
         {
            "access":3,
            "description":"Comfort temperature",
            "name":"comfort_temperature",
            "property":"comfort_temperature",
            "type":"numeric",
            "unit":"°C"
         },
         {
            "access":3,
            "description":"Eco temperature",
            "name":"eco_temperature",
            "property":"eco_temperature",
            "type":"numeric",
            "unit":"°C"
         },
         {
            "access":3,
            "description":"Force the valve position",
            "name":"force",
            "property":"force",
            "type":"enum",
            "values":[
               "normal",
               "open",
               "close"
            ]
         },
         {
            "access":3,
            "description":"Maximum temperature",
            "name":"max_temperature",
            "property":"max_temperature",
            "type":"numeric",
            "unit":"°C"
         },
         {
            "access":3,
            "description":"Minimum temperature",
            "name":"min_temperature",
            "property":"min_temperature",
            "type":"numeric",
            "unit":"°C"
         },
         {
            "access":3,
            "description":"Week format user for schedule",
            "name":"week",
            "property":"week",
            "type":"enum",
            "values":[
               "5+2",
               "6+1",
               "7"
            ]
         },
         {
            "access":3,
            "description":"Away preset temperature",
            "name":"away_preset_temperature",
            "property":"away_preset_temperature",
            "type":"numeric",
            "unit":"°C"
         },
         {
            "access":1,
            "description":"Link quality (signal strength)",
            "name":"linkquality",
            "property":"linkquality",
            "type":"numeric",
            "unit":"lqi",
            "value_max":255,
            "value_min":0
         }
      ],
      "model":"TS0601_thermostat",
      "supports_ota":true,
      "vendor":"TuYa"
   },
   "endpoints":{
      "1":{
         "bindings":[
            
         ],
         "clusters":{
            "input":[
               "genBasic",
               "genGroups",
               "genScenes",
               "manuSpecificTuya"
            ],
            "output":[
               "genOta",
               "genTime"
            ]
         },
         "configured_reportings":[
            
         ]
      }
   },
   "friendly_name":"radiator Lieke",
   "ieee_address":"0x847127fffe11dc50",
   "interview_completed":true,
   "interviewing":false,
   "model_id":"TS0601",
   "network_address":11042,
   "power_source":"Battery",
   "supported":true,
   "type":"EndDevice"
}
```

**Comment by Koenkk** [comment]
The exposes switch is not only used to describe something that can be physically turned off/on but is also meant for turning on/off certain options. We can consider moving it into climate but I'm wondering if this actually improves something.

**Comment by itavero** [comment]
The main thing for me is that there are multiple "switches" within the same endpoint, which only differ on their `property` (and description, but that's not something I can use programmatically anyway).

I don't think the `property` of a feature should be used to distinguish between multiple items with a composite type (`switch` in this case). It feels a bit like a hack, to traverse down the tree that far to determine the uniqueness of an item in the exposes array.

In other words, I'm not really against having these "switches" as long as they are somehow unique on the "top" level of the exposes information (like when an `endpoint` is specified).
Perhaps there should be an additional field on the "top" level in case multiple instances of the same `type` and `endpoint` exist in a single model.

**Comment by Koenkk** [comment]
> Perhaps there should be an additional field on the "top" level

How should this property be named and what should be put into it?

**Comment by itavero** [comment]
I'll try to scan through the other definitions in `devices.js` to see if there are more cases in which the exposes array contains entries that are pretty much identical.
Based on that I'll see if a can come up with a decent proposal.
